agent/src/beerocks/slave/gate/1905_beacon_query_to_vs.cpp

```diff
@@ -0,0 +1,115 @@
+
+#include "1905_beacon_query_to_vs.h"
+#include <bcl/beerocks_defines.h>
+#include <mapf/common/utils.h>
+
+namespace beerocks {
+namespace gate {
+
+bool load(std::shared_ptr<beerocks_message::cACTION_MONITOR_CLIENT_BEACON_11K_REQUEST> lhs,
+          const ieee1905_1::CmduMessageRx &rhs)
+{
+    // escape I
+    if (!lhs) {
+        LOG(ERROR) << "can't fill empty pointer. lhs is nullptr. returning";
+        return false;
+    }
+
+    // escape II
+    // get the correct type of the message
+    auto beacon_metrics_query = rhs.getClass<wfa_map::tlvBeaconMetricsQuery>();
+    if (!beacon_metrics_query) {
+        LOG(ERROR) << "tlvBeaconMetricsQuery is missing - can't read from invalid input";
+        return false;
+    }
+
+    auto &lhs_params = lhs->params();
+
+    lhs_params.bssid   = beacon_metrics_query->bssid();
+    lhs_params.channel = beacon_metrics_query->channel_number();
+
+    lhs_params.measurement_mode = beerocks::MEASURE_MODE_ACTIVE;
+    lhs_params.duration         = beerocks::BEACON_MEASURE_DEFAULT_ACTIVE_DURATION;
+
+    lhs_params.expected_reports_count = 1;
+
+    lhs_params.rand_ival = beerocks::BEACON_MEASURE_DEFAULT_RANDOMIZATION_INTERVAL;
+    lhs_params.sta_mac   = beacon_metrics_query->associated_sta_mac();
+
+    lhs_params.op_class = beacon_metrics_query->operating_class();
+
+    // FixMe: not sure about this
+    lhs_params.report = beacon_metrics_query->reporting_detail_value();
+
+    // values based on https://github.com/prplfoundation/prplMesh/pull/1114#discussion_r406326546
+    lhs_params.repeats            = 0;
+    lhs_params.parallel           = 0;
+    lhs_params.enable             = 0;
+    lhs_params.request            = 1;
+    lhs_params.mandatory_duration = 0;
+    lhs_params.use_optional_ssid  = 0;
+
+    mapf::utils::copy_string(lhs_params.ssid, beacon_metrics_query->ssid_str().c_str(),
+                             beerocks::message::WIFI_SSID_MAX_LENGTH);
+
+    // how many elements in the list of channels, we support only 1 at the moment
+    auto ap_ch_report_length = beacon_metrics_query->ap_channel_reports_list_length();
+
+    // escape III - data validation
+    // ap_ch_report_length != 0 ==> channel_number == 255
+    // we verify with the already extracted values (we could do it withthe original as well)
+    if (ap_ch_report_length != 0 && lhs_params.channel != 255) {
+        LOG(ERROR) << "inconsistency between channel report length and channel number. please take "
+                      "look at the specification. v1 17.2.27";
+        return false;
+    }
+
+    // copy m_ap_channel_reports_list from the 1905 beacon_metrics_query
+    // into ap_ch_report char array of the vs message (lhs_params)
+    if (ap_ch_report_length > 1) {
+        // the reason we support only 1 channel list is that the hostapd does not support more
```

Comment:
Actually it's the VS message that doesn't support more. And Intel's hacked-up hostapd (AKA dwpal). Upstream hostapd supports it (because you actually have to give it the complete beacon request frame).